src/Symfony/Bundle/FrameworkBundle/Tests/Command/TranslationDebugCommandTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Tests\Command;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Bundle\FrameworkBundle\Command\TranslationDebugCommand;
+use Symfony\Component\Filesystem\Filesystem;
+
+class TranslationDebugCommandTest extends \PHPUnit_Framework_TestCase
+{
+    private $fs;
+    private $translationDir;
+
+    public function testDebugMissingMessages()
+    {
+        $tester = $this->createCommandTester($this->getContainer(array('foo' => 'foo')));
+        $tester->execute(array('locale' => 'en', 'bundle' => 'foo'));
+
+        $this->assertRegExp('/x (\s|\|)+foo/', $tester->getDisplay(), 'Display x in case of missing message');
+    }
+
+    public function testDebugUnusedMessages()
+    {
+        $tester = $this->createCommandTester($this->getContainer(array(), array('foo' => 'foo')));
+        $tester->execute(array('locale' => 'en', 'bundle' => 'foo'));
+
+        $this->assertRegExp('/o (\s|\|)+foo/', $tester->getDisplay(), 'Display o in case of unused message');
+    }
+
+    public function testDebugFallbackMessages()
+    {
+        $tester = $this->createCommandTester($this->getContainer(array(), array('foo' => 'foo')));
+        $tester->execute(array('locale' => 'fr', 'bundle' => 'foo'));
+
+        $this->assertRegExp('/= (\s|\|)+foo/', $tester->getDisplay(), 'Display = in case of fallback message');
+    }
+
+    public function testNoDefinedMessages()
+    {
+        $tester = $this->createCommandTester($this->getContainer());
+        $tester->execute(array('locale' => 'fr', 'bundle' => 'test'));
+
+        $this->assertRegExp('/^No defined or extracted messages for locale "fr"/', $tester->getDisplay());
+    }
+
+    protected function setUp()
+    {
+        $this->fs = new Filesystem();
+        $this->translationDir = sys_get_temp_dir().DIRECTORY_SEPARATOR.uniqid('sf2_translation');
+        $this->fs->mkdir($this->translationDir.DIRECTORY_SEPARATOR.'Resources');
+        $this->fs->mkdir($this->translationDir.DIRECTORY_SEPARATOR.'Resources'.DIRECTORY_SEPARATOR.'translations');
```

Comment:
use ``/Resources/translations``. It works well on PHP 5.3+ and makes things much more readable (we use ``DIRECTORY_SEPARATOR`` only when we need to perform string comparisons on paths to avoid issues on Windows)